connectors/src/connectors/notion/temporal/signals.ts

```diff
@@ -1,10 +1,63 @@
 import { defineSignal } from "@temporalio/workflow";
+import { z } from "zod";
+
+import logger from "@connectors/logger/logger";
 
 export type NotionWebhookEvent = {
-  type: string;
+  type: "database.deleted" | "page.deleted";
   entity_id: string;
 };
 
 export const notionWebhookSignal = defineSignal<[NotionWebhookEvent]>(
   "notion_webhook_signal"
 );
+
+export type NotionDeletionCrawlSignal = {
+  resourceId: string;
+  resourceType: "page" | "database";
+};
+
+export const notionDeletionCrawlSignal = defineSignal<
+  [NotionDeletionCrawlSignal]
+>("notion_deletion_crawl_signal");
+
+/**
+ * Zod schema for validating deletion crawl signal arguments
+ * Validates connectorId, resourceId, and resourceType together
+ */
+export const DeletionCrawlSignalArgsSchema = z.object({
```

Comment:
Maybe a bit overkill to use this for something we fully control? I think we typically use this to validate external data.